### PR TITLE
docs(guides/configuration): Fine-tune wording in configuration guide

### PR DESCRIPTION
**Pull request by PascalPrecht**

### CODE REVIEW

File: source/docs/configuration.md [hunk 1]
```diff
@@ -12,62 +19,72 @@ title: Configuring Embark
...
+
 ### contracts
 
-List of directories in which embark should look for contracts. These typically are GLOB expressions (e.g `["contracts/**/*.sol"]` will match all sol files inside any folders inside `contracts/`)
+This is a list of directories in which Embark should look for Smart Contract files. These typically are globbing patterns (e.g `["contracts/**/*.sol"]` will match all sol files inside any folders inside `contracts/`).
+
+### app
 
-### pipeline
+Everything inside `app` configures the assets of our application. At the moment there are three types of assets to configure: JavaScript files, HTML and images.
```
Comment (jrainville) [inline source/docs/configuration.md]: Not 100% true. It works for CSS files and all other assets that need to be simply copied.

Comment (michaelsbradleyjr) [reply source/docs/configuration.md]: Keys under `"app"` that end in `.js` are basically output files for webpack, and the files in the arrays corresponding to those `.js` keys are each an entry point for webpack that will be packed (along with their respective dependencies) and then included together in the respective output file. Keys that don't end in `.js`, whether a directory or single file, are copy-targets for the file/s / glob-patterns corresponding to those keys — the contents aren't run through webpack, simply copied as @jrainville says.

Comment (PascalPrecht) [reply source/docs/configuration.md]: Okay, let me reword this.

Comment (PascalPrecht) [reply source/docs/configuration.md]: Updated!

File: source/docs/configuration.md [hunk 2]
```diff
@@ -12,62 +19,72 @@ title: Configuring Embark
...
+
+### app
 
-### pipeline
+Everything inside `app` configures the assets of our application, which could be anything from HTML, JavaScript to CSS and other assets.
 
-The mapping between the output files and the source. e.g `"js/dapp.js": ["app/dapp.js"]` will run `app/dapp.js` though the embark pipeline, and output the result into `dist/js/dapp.js`
+  - **js/dapp.js** - This is the JavaScript bundle that contains our application. Specifically, all files that are defined in this option (`app/dapp.js`).
+  - **index.html** - The entry point of our application (`app/index.html`)
+  - **images** - All image assets of our application that we can find in `app/images/`.
```
Comment (jrainville) [inline source/docs/configuration.md]: I like the explanations you put here, but I feel we could go a bit more general. Something like "JS files are compiled using Webpack to create a bundle, all other files types are simply copied."
It's just that with your explanation, it looks like the above is the only way to go.
Also, it's important to explain that the left-hand side is the destination and the right-hand side is the starting files. Also, that we can use globs.
Sorry if I worded everything badly. I'm not sure how we should put it up.

**Comment by PascalPrecht** [comment]
@jrainville alright I've updated this one once more. Care taking another look? Let me know if this needs further change! :)

File: source/docs/configuration.md [hunk 3]
```diff
@@ -12,62 +19,74 @@ title: Configuring Embark
...
 
-List of directories in which embark should look for contracts. These typically are GLOB expressions (e.g `["contracts/**/*.sol"]` will match all sol files inside any folders inside `contracts/`)
+This is a list of directories in which Embark should look for Smart Contract files. These typically are globbing patterns (e.g `["contracts/**/*.sol"]` will match all sol files inside any folders inside `contracts/`).
+
+### app
+
+Everything inside `app` configures the assets of our application, which could be anything from HTML, JavaScript to CSS and other assets. JavaScript source files are compiled using webpack to create a bundle, all other file types are simply copied to the specified destination.
 
-### pipeline
+In the key/value pairs of `app` related configurations, every key descriptions the destination, while the value describes a list of glob patterns for files to be transformed and moved.
```
Comment (jrainville) [inline source/docs/configuration.md]:
```suggestion
In the key/value pairs of `app`, every key describes the destination, while the value describes a list of glob patterns for files to be transformed and moved.
```

Comment (michaelsbradleyjr) [reply source/docs/configuration.md]: Maybe "copied" instead of "moved"?

**Comment by PascalPrecht** [comment]
Updated :)

**Pull request merged**